crates/ty_python_semantic/src/types/class_base.rs

```diff
@@ -170,7 +170,8 @@ impl<'db> ClassBase<'db> {
                 | KnownInstanceType::TypeVar(_)
                 | KnownInstanceType::Deprecated(_)
                 | KnownInstanceType::Field(_)
-                | KnownInstanceType::ConstraintSet(_) => None,
+                | KnownInstanceType::ConstraintSet(_)
+                | KnownInstanceType::UnionType(_) => None,
```

Comment:
This is different from the `Type::Union` branch above. `UnionType` instances are not allowed as bases.

Reply:
can we add an explicit test for that?